### ISSUE TITLE
Possible naming collision with sysprop `gradle.cache.remote.url`

### ISSUE THREAD

**Issue by DPUkyle** [gradle/common-custom-user-data-gradle-plugin#141]
As of v1.9, the presence of `systemProp.gradle.cache.remote.url=https://us-build-cache.gradle.org/cache/` in `~/.gradle/gradle.properties` causes this failure:

```
FAILURE: Build failed with an exception.

* What went wrong:
The Gradle Enterprise server address 'https://us-build-cache.gradle.org/cache/' value should not contain a path component

* Try:
> Run with --debug option to get more log output.

* Exception is:
java.lang.IllegalArgumentException: The Gradle Enterprise server address 'https://us-build-cache.gradle.org/cache/' value should not contain a path component
        at com.gradle.scan.agent.a.b.c.a(SourceFile:63)
        at com.gradle.scan.agent.a.b.c.a(SourceFile:54)
        at com.gradle.enterprise.gradleplugin.GradleEnterpriseBuildCache.setServer(SourceFile:82)
        at com.gradle.enterprise.gradleplugin.internal.a.b_Decorated.setServer(Unknown Source)
        at com.gradle.Overrides.lambda$configureBuildCache$6(Overrides.java:76)
        at org.gradle.caching.configuration.internal.DefaultBuildCacheConfiguration.remote(DefaultBuildCacheConfiguration.java:101)
        at com.gradle.Overrides.configureBuildCache(Overrides.java:75)
        at com.gradle.CommonCustomUserDataGradlePlugin.lambda$applySettingsPlugin$0(CommonCustomUserDataGradlePlugin.java:76)
...
```

Steps to reproduce:

1. `$ git pull # HEAD is 77367a034f3330d6c7ade7094b51c8fdc2158961`
2. `$ ./gradlew clean -is # results in error`

To workaround, apply this patch:

```
diff --git a/settings.gradle b/settings.gradle
index fe344f0..f31e331 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -1,6 +1,6 @@
 plugins {
     id 'com.gradle.enterprise' version '3.12.4'
-    id 'com.gradle.common-custom-user-data-gradle-plugin' version '1.9'
+    id 'com.gradle.common-custom-user-data-gradle-plugin' version '1.8'
 }
 
 def isCI = System.getenv('GITHUB_ACTIONS') != null
```

**Comment by DPUkyle** [comment]
Note this was originally detected in gradle/gradle-enterprise-build-validation-scripts#358

**Comment by etiennestuder** [comment]
This is not related to the scripts but only to the CCUD plugin. The script does not do anything with that cache property. All build configuration done by the scripts use the prefix `com.gradle.enterprise.build-validation`.

**Comment by etiennestuder** [comment]
Apologies, I thought this issue was created on the scripts repo.

**Comment by etiennestuder** [comment]
@DPUkyle this is only an issue for people working with both the BT code base and the CCUD plugin. Given this affects only very few people (and no customers), I'm very much in favor of the BT build using a different property to configure the build cache in their build.

**Comment by etiennestuder** [comment]
> I'm trying to understand this problem. Since CCUD plugin 1.9, it starts reading system property `gradle.cache.remote.url`, and expects it to be GE host url, right? 

No.

> Now I'm a bit confused, the property `gradle.cache.remote.url` is recognized as "Gradle Enterprise server address", why we mixed up "cache url" and "GE server url"?

We don't.

- that property has been in use by the CCUD plugin since November 2021
- nothing has changed in the behavior with CCUD plugin 1.9
- that property is not used to set the GE URL but the remote cache URL. see [here](https://github.com/gradle/common-custom-user-data-gradle-plugin/blob/8607224807b623076dd073197ded3234d45b8656/src/main/java/com/gradle/Overrides.java#L65-L83)

What we could do is, in the case of the GradleEnterpriseBuildCache, split the remote cache URL provided via sys prop in `server` and `path` and set these individually on the 2 APIs on the GradleEnterpriseBuildCache.

**Comment by etiennestuder** [comment]
> Now I understand after reading the code. Yes, that would be the best solution instead of throwing a confusing exception.

The exception message comes from the GE plugin.

**Comment by etiennestuder** [comment]
@DPUkyle Thanks for the report. This will be fixed with the following PR, once CCUD plugin 1.9.1 is released.
https://github.com/gradle/common-custom-user-data-gradle-plugin/pull/142

**Comment by etiennestuder** [comment]
This has been addressed more thoroughly in #146 and will be released in CCUD 1.10 (there will be no 1.9.1).

@DPUkyle @blindpirate You might run into the same issue with any other plugin (or init script) that claims the `gradle.cache.remote.url` property if you configure its value in gradle.properties in the user home and expect it to only be picked up by the Gradle Build Tool convention plugin.  IMO, a more robust approach would be to have an init script in the user home that configures the remote build cache URL only for a configurable set of projects that you want to include. I believe this is what Cedric used to do successfully.

**Comment by etiennestuder** [comment]
Released: https://github.com/gradle/common-custom-user-data-gradle-plugin/releases/tag/v1.10
